warden/setup-environment/action.yml

```diff
@@ -0,0 +1,398 @@
+        WARDEN_REDIS=${REDIS}
+
+        ELASTICSEARCH_VERSION=${ELASTICSEARCH_VERSION}
+        OPENSEARCH_VERSION=${OPENSEARCH_VERSION}
+        DB_DISTRIBUTION=${DB_DISTRIBUTION}
+        DB_DISTRIBUTION_VERSION=${DB_DISTRIBUTION_VERSION}
+        NODE_VERSION=12
+        COMPOSER_VERSION=${COMPOSER_VERSION}
+        PHP_VERSION=${PHP_VERSION}
+        PHP_XDEBUG_3=1
+        RABBITMQ_VERSION=${RABBITMQ_VERSION}
+        REDIS_VERSION=${REDIS_VERSION}
+        VARNISH_VERSION=${VARNISH_VERSION}
+
+        WARDEN_SYNC_IGNORE=
+
+        WARDEN_ALLURE=0
+        WARDEN_SELENIUM=0
+        WARDEN_SELENIUM_DEBUG=0
+        WARDEN_BLACKFIRE=0
+        WARDEN_SPLIT_SALES=0
+        WARDEN_SPLIT_CHECKOUT=0
+        WARDEN_TEST_DB=1
+        WARDEN_MAGEPACK=0
+
+        BLACKFIRE_CLIENT_ID=
+        BLACKFIRE_CLIENT_TOKEN=
+        BLACKFIRE_SERVER_ID=
+        BLACKFIRE_SERVER_TOKEN=
+        EOF
+
+        echo ".env created"
+        cat .env
+
+    - name: Warden svc up && Warden env up
+      working-directory: ${{ inputs.base_directory }}
+      shell: bash
+      run: |
+          export WARDEN="$(dirname $(pwd))/warden/bin/warden"
+          ${WARDEN} svc up 
+          ${WARDEN} env up
+
+    - name: Change Directory Permissions
+      working-directory: ${{ inputs.base_directory }}
+      shell: bash
+      run: |
+        export WARDEN="$(dirname $(pwd))/warden/bin/warden"
+        ${WARDEN} env exec -T php-fpm sudo chmod -R 777 .
+
+    - name: Wait for environment to be ready
+      working-directory: ${{ inputs.base_directory }}
+      shell: bash
+      env:
+        SEARCH: ${{ inputs.search }}
+      run: |
+        export WARDEN="$(dirname $(pwd))/warden/bin/warden"
+        case "${SEARCH}" in
+          elasticsearch:*)
+            SEARCH_HOST="elasticsearch"
+            ;;
+          *)
+            SEARCH_HOST="opensearch"
+            ;;
+        esac
+        HEALTHY=1
+        for ((i=1; i<=24; i++)); do
+          SEARCH_STATUS=$(${WARDEN} env exec -T php-fpm bash -c "curl --write-out %{http_code} --silent --output /dev/null http://${SEARCH_HOST}:9200/_cat/health?h=st; exit 0")
+          echo "search status: ${SEARCH_STATUS}"
+          if [ ${SEARCH_STATUS} -eq "200" ]; then
+            HEALTHY=0
+            break
+          fi
+          sleep 5
+        done
+        echo "HEALTHY: ${HEALTHY}"
+        exit ${HEALTHY}
+        
+    - name: composer install run
+      working-directory: ${{ inputs.base_directory }}
+      shell: bash
+      env:
+        RUN_INSTALL: ${{ inputs.run_composer_install }}
+      run: |
+        if [ $RUN_INSTALL == "1" ]; then
+          export WARDEN="$(dirname $(pwd))/warden/bin/warden"
+          wget https://getcomposer.org/download/${{ inputs.composer_version }}/composer.phar -O composer.phar 1>/dev/null 2>&1 && chmod +x composer.phar
+          ${WARDEN} env exec -T php-fpm ls -la | grep composer
+          ${WARDEN} env exec -T php-fpm php composer.phar -v
```

Comment:
Do we want / need this?

Reply:
I think so, because I like explicit debug information for implicit things :)  
Long story short - warden does not support exact composer version usage. Composer 1 or 2 - that is the only thing you can get out of it. Even if you put in .env file `COMPOSER_VERSION=2.2` it won't work, it will even give you composer v1 instead 😁  
Seems there is a bug, or at least was a month ago when I was adding this.  

That is why we are downloading and using composer.phar of exact version we are defining from the workflow. And me as a developer would be happy to see that I've downloaded this exact version while checking out why pipeline does not work (if it will not gonna work at some point of time).  

My thoughts.

Reply:
I think we leave in for now, and see how it's used?
It might be worth swapping to a docker container instead like the other actions do. But that would stop people extending / re-using this action if they are expecting composer to be installed.

Reply:
Fine for me :+1: